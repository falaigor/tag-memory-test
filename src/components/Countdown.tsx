import { useState, useEffect } from "react";

interface CountdownProps {
  startCountdown: boolean;
}

export const Countdown = ({ startCountdown }: CountdownProps) => {
  let countdownTimeOut;
  const [time, setTime] = useState(1 * 60);

  const minutes = Math.floor(time / 60);
  const seconds = time % 60;

  const [minuteLeft, minuteRight] = String(minutes)
    .padStart(2, "0")
    .split("");
  const [secondLeft, secondRight] = String(seconds)
    .padStart(2, "0")
    .split("");

  useEffect(() => {
    if (startCountdown === true && time > 0) {
      countdownTimeOut = setTimeout(() => {
        setTime(time - 1);
      }, 1000);
    }
  }, [time, startCountdown]);

  return (
    <div className="flex flex-row font-montserrat text-7xl">
      <div>
        <span>{minuteLeft}</span>
        <span>{minuteRight}</span>
      </div>
      <span>:</span>
      <div>
        <span>{secondLeft}</span>
        <span>{secondRight}</span>
      </div>
    </div>
  );
};
