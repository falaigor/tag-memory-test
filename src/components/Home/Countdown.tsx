import { useChallenge } from "../../contexts/challenge";

export const Countdown = () => {
  const { time } = useChallenge();
  const minutes = Math.floor(time / 60);
  const seconds = time % 60;

  const [minuteLeft, minuteRight] = String(minutes).padStart(2, "0").split("");
  const [secondLeft, secondRight] = String(seconds).padStart(2, "0").split("");

  return (
    <div
      data-testid="countdown"
      className="flex flex-row font-montserrat text-7xl"
    >
      <div data-testid="minute">
        <span>{minuteLeft}</span>
        <span>{minuteRight}</span>
      </div>
      <span>:</span>
      <div data-testid="second">
        <span>{secondLeft}</span>
        <span>{secondRight}</span>
      </div>
    </div>
  );
};
