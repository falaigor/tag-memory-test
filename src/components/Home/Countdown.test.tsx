import { render, screen } from "../../tests/test-utils";
import { Countdown } from "./Countdown";

describe("Countdown", () => {
  it("should render Countdown component", () => {
    render(<Countdown />);

    expect(screen.getByTestId("countdown")).toBeInTheDocument();
  });
});
