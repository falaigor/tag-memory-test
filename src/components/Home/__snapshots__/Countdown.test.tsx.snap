// Vitest Snapshot v1

exports[`Countdown > should show a minutes 1`] = `
"
    <div
    data-testid=\\"minute\\"
  >
    <span>
      0
    </span>
    <span>
      1
    </span>
  </div>
    "
`;

exports[`Countdown > should show a seconds 1`] = `
"
    <div
    data-testid=\\"second\\"
  >
    <span>
      0
    </span>
    <span>
      5
    </span>
  </div>
    "
`;
